experimental: RPA documentation (#4440)
camunda/camunda-docs: docs/components/experimental/rpa/rpa-integration-with-camunda.md

---
id: rpa-integration
title: RPA integration
description: "Learn how Camunda enables orchestration of simple robotic process automation (RPA) tasks within Camunda 8 based on the RPA Framework."
---

Camunda enables orchestration of simple **robotic process automation (RPA)** tasks within Camunda 8 based on the [RPA Framework](https://rpaframework.org/). The RPA integration consists of two primary components: RPA runtime and the [Camunda Desktop Modeler](/components/modeler/about-modeler.md) plugin.

## RPA runtime

> The RPA runtime facilitates the execution of your RPA scripts and is suitable for both local testing and script execution from your Camunda SaaS [cluster](//guides/create-cluster.md).

Reviewer: ```suggestion
You can use the RPA runtime to run your RPA scripts for both local testing or from your Camunda SaaS [cluster](//guides/create-cluster.md).
```
Non-blocking suggestion for conciseness - bigger topic is execute vs run 😄

- **For a full development setup**: Visit the [Camunda Prototype RPA Worker GitHub repository](https://github.com/camunda/rpa-runtime).
- **For a Windows setup without Python**: Take the following steps:
  1. Download the [RPA runner repository](https://github.com/camunda/rpa-runtime/archive/refs/heads/main.zip) from GitHub.
  2. Unzip the file and navigate to the `rpa-runtime-main` repository folder.
  3. Execute `setup.ps1` using PowerShell. If permission issues arise, alter your Execution Policy with the command `Set-ExecutionPolicy RemoteSigned`.

>   4. Enter your credentials into the .env file. Refer to [the documentation on how to create credentials](/components/console/manage-clusters/manage-api-clients.md#create-a-client).

Reviewer: ```suggestion
  4. Enter your credentials into the `.env` file. See [how to create credentials](/components/console/manage-clusters/manage-api-clients.md#create-a-client).
```
Non-blocking suggestion: Should the file be in code?

- `/workers/{scriptName}/{jobId}/`: Stores logs and runtime data for your workers, where the job ID corresponds to the Zeebe job ID when executing tasks from Camunda.

## Camunda Desktop Modeler plugin

:::note
Desktop Modeler version 5.29.0 or higher is required for this plugin. Until its release on November 8th 2024, use the [nightly build](https://downloads.camunda.cloud/release/camunda-modeler/nightly/).
:::

### Installation

For more information on plugin installation, visit the [plugin installation guide](/components/modeler/desktop-modeler/plugins/plugins.md).

To install the RPA plugin, take the following steps:

1. Download the latest version of the [RPA plugin](http://github.com/camunda/camunda-modeler-rpa-plugin/zipball/latest/) from GitHub.
2. Extract the folder to `resources/plugins` within your Desktop Modeler installation directory.
3. Ensure the path is structured correctly: `/resources/plugins/<plugin-name>/index.js`.
4. Launch or restart Desktop Modeler.

You can now open and edit `.robot` files in Desktop Modeler.

### Robot development

To create or edit a `.robot` file, use the welcome page or file dialog in Modeler.

> The editor allows for script development, with all RPA Framework libraries listed [here](https://rpaframework.org/#libraries). Camunda also offers a [Camunda library extension](/components/experimental/rpa/camunda-rpa-framework-library.md) to integrate with Camunda and return variables from your RPA scripts, for example.

Reviewer: ```suggestion
The editor allows for script development, with all RPA Framework libraries listed in the [RPA Framework documentation](https://rpaframework.org/#libraries). Camunda also offers a [Camunda library extension](/components/experimental/rpa/camunda-rpa-framework-library.md) to integrate with Camunda and return variables from your RPA scripts, for example.
```

Currently, only a single `.robot` file is supported, and adding custom Python files or libraries is not possible.

### Testing

> To test your script, select the “Play” icon in the bottom left corner of Desktop Modeler to access the **Robot Testing** tab.

Reviewer: ```suggestion
To test your script, select the **Play** icon in the bottom left corner of Desktop Modeler to access the **Robot Testing** tab.
```
Non-blocking suggestion

![RPA testing example](../img/testing-rpa.png)

> You can now run the script with the expected input from the process on your local RPA runtime. Once the script finishes, you can see a more detailed view of the run by selecting the **Show log** button.

Reviewer: ```suggestion
You can now run the script with the expected input from the process on your local RPA runtime. Once the script completes, select **Show log** to see a more detailed view of the run.
```
Non-blocking suggestion

---
id: rpa-integration-with-camunda
title: RPA integration with Camunda
description: "Learn how Camunda enables orchestration of simple robotic process automation (RPA) tasks within Camunda 8 based on the RPA Framework."
---

Camunda enables orchestration of simple **robotic process automation (RPA)** tasks within Camunda 8 based on the [RPA Framework](https://rpaframework.org/). The RPA integration consists of two primary components: RPA runtime and the Camunda Desktop Modeler plugin.

## RPA runtime

The RPA runtime facilitates the execution of your RPA scripts and is suitable for both local testing and script execution from your Camunda SaaS cluster.

- **For a full development setup**: Visit the [Camunda Prototype RPA Worker GitHub repository](https://github.com/camunda/rpa-runtime).

> - **For a Windows setup without Python**: Take the following steps:
> 
> 1. Download the [RPA runner repository](https://github.com/camunda/rpa-runtime/archive/refs/heads/main.zip) from GitHub.
> 2. Unzip the file and navigate to the `rpa-runtime-main` repository folder.
> 3. Execute `setup.ps1` using PowerShell. If permission issues arise, alter your Execution Policy with the command `Set-ExecutionPolicy RemoteSigned`.
> 4. Enter your credentials into the .env file. Refer to [the documentation on how to create credentials](/components/console/manage-clusters/manage-api-clients.md#create-a-client).
> 5. Start the RPA worker locally by running `start.ps1` in PowerShell. To stop the worker, close the PowerShell terminal.

Reviewer 2: Make this a sublist
```suggestion
- **For a Windows setup without Python**: Take the following steps:
  1. Download the [RPA runner repository](https://github.com/camunda/rpa-runtime/archive/refs/heads/main.zip) from GitHub.
  2. Unzip the file and navigate to the `rpa-runtime-main` repository folder.
  3. Execute `setup.ps1` using PowerShell. If permission issues arise, alter your Execution Policy with the command `Set-ExecutionPolicy RemoteSigned`.
  4. Enter your credentials into the .env file. Refer to [the documentation on how to create credentials](/components/console/manage-clusters/manage-api-clients.md#create-a-client).
  5. Start the RPA worker locally by running `start.ps1` in PowerShell. To stop the worker, close the PowerShell terminal.
```
